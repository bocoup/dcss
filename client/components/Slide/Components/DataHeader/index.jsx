import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { Form, Message, Popup } from 'semantic-ui-react';

class DataHeader extends Component {
  render() {
    const { content, onChange } = this.props;

    const trigger = (
      <Fragment>
        <Form.TextArea
          required
          label="Data Header"
          name="header"
          value={content}
          onChange={onChange}
        />
        This is only displayed in researcher data, as the column name for this
        response.
      </Fragment>
    );
    const popup = (
      <Popup
        content="Set a data header. This is only displayed in the data view and data download."
        trigger={trigger}
      />
    );
    return <Message color={content ? 'grey' : 'red'} content={popup} />;
  }
}

DataHeader.propTypes = {
  onChange: PropTypes.func.isRequired,
  content: PropTypes.any
};

export default DataHeader;
