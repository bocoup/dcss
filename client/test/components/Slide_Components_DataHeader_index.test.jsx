import {
  fetchImplementation,
  mounter,
  reduxer,
  snapshot,
  state,
} from '../bootstrap';
import {
  unmountComponentAtNode
} from 'react-dom';
import {
  cleanup,
  fireEvent,
  render
} from '@testing-library/react';
import DataHeader from '../../components/Slide/Components/DataHeader/index.jsx';

beforeAll(() => {
  (window || global).fetch = jest.fn();
});

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  container.setAttribute('id', 'root');
  document.body.appendChild(container);

  fetchImplementation(fetch);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

const sharedProps = {
  history: {
    push() {},
  },
};

test('DataHeader', () => {
  expect(DataHeader).toBeDefined();
});

test('Snapshot 1', () => {
  const props = {
    ...sharedProps,
  };
  const mounted = mounter(reduxer(DataHeader, props, state))();
  expect(snapshot(mounted)).toMatchSnapshot();
});

/*{INJECTION}*/
