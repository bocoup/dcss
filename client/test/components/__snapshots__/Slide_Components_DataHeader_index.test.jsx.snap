// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1 1 1`] = `"<div class=\\"ui red message\\"><div class=\\"content\\"><div class=\\"required field\\"><label>Data header</label><div class=\\"ui input\\"><input name=\\"header\\" required=\\"\\" type=\\"text\\" value=\\"\\"></div></div>This will only appear in data downloads, as the column header or row name.</div></div>"`;

exports[`Snapshot 1 2 1`] = `"<div class=\\"ui red message\\"><div class=\\"content\\"><div class=\\"required field\\"><label>Data header</label><div class=\\"ui input\\"><input name=\\"header\\" required=\\"\\" type=\\"text\\" value=\\"\\"></div></div>This will only appear in data downloads, as the column header or row name.</div></div>"`;
